cmd/2019_7/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"flag"
+
+	"github.com/afarbos/aoc/pkg/intcode"
+	"github.com/afarbos/aoc/pkg/mathematic"
+	"github.com/afarbos/aoc/pkg/perm"
+	"github.com/afarbos/aoc/pkg/read"
+	"github.com/afarbos/aoc/pkg/utils"
+)
+
+var flagInput string
+
+func init() {
+	utils.Init(&flagInput)
+}
+
+func findMaxThrusterSignalWithSettings(instructions []int, settingsSet map[int]struct{}) int {
+	signal := mathematic.MaxInt()
+
+	perm.Ints(settingsSet, func(settings []int) {
+		var (
+			inputs = make([](chan int), 0, len(settings))
+		)
+
+		for range settings {
+			inputs = append(inputs, make(chan int, 1))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [avoid magic numbers like '1', create a named constant for it](https://app.codacy.com/manual/afarbos/aoc/pullRequest?prid=5498712)